### PR TITLE
Holocene 1559 params for block building

### PR DESCRIPTION
**Pull request by mininny**
This PR includes changes required to implement the configurable EIP-1559 parameters that are part of the Holocene upgrade.

Prerequisite erigon-interfaces PR: https://github.com/testinprod-io/erigon-interfaces/pull/21
Spec: ethereum-optimism/specs#340
Mirrored from op-geth PR: https://github.com/ethereum-optimism/op-geth/pull/398

### CODE REVIEW

File: consensus/misc/eip1559.go [hunk 1]
```diff
@@ -101,18 +102,55 @@ func (f eip1559Calculator) CurrentFees(chainConfig *chain.Config, db kv.Getter)
 	return baseFee, blobFee, minBlobGasPrice, currentHeader.GasLimit, nil
 }
 
-// CalcBaseFee calculates the basefee of the header.
+// DecodeHolocene1599Params extracts the Holcene 1599 parameters from the encoded form:
```
Comment (pcw109550) [inline consensus/misc/eip1559.go]:
```suggestion
// DecodeHolocene1559Params extracts the Holcene 1559 parameters from the encoded form:
```

We need to fix 1599 to 1559 throughout the code.

File: turbo/execution/eth1/block_building.go [hunk 2]
```diff
@@ -57,6 +57,7 @@ func (e *EthereumExecutionModule) AssembleBlock(ctx context.Context, req *execut
 		Transactions:          req.Transactions,
 		NoTxPool:              req.NoTxPool,
 		GasLimit:              req.GasLimit,
+		EIP1559Params:         req.Eip_1599Params,
```
Comment (pcw109550) [inline turbo/execution/eth1/block_building.go]:
```suggestion
		EIP1559Params:         req.Eip_1559Params,
```

File: turbo/engineapi/engine_server.go [hunk 3]
```diff
@@ -530,6 +531,7 @@ func (s *EngineServer) forkchoiceUpdated(ctx context.Context, forkchoiceState *e
 		SuggestedFeeRecipient: gointerfaces.ConvertAddressToH160(payloadAttributes.SuggestedFeeRecipient),
 		Transactions:          txs,
 		NoTxPool:              payloadAttributes.NoTxPool,
+		Eip_1599Params:        payloadAttributes.EIP1559Params,
```
Comment (pcw109550) [inline turbo/engineapi/engine_server.go]:
```suggestion
		Eip_1559Params:        payloadAttributes.EIP1559Params,
```

File: consensus/misc/eip1559.go (hunk 1 above)
Comment (mininny) [reply consensus/misc/eip1559.go]: I fixed it throughout the whole code! :)

**Review by pcw109550 — APPROVED** [review]

**Pull request merged**